### DISCUSSION TITLE
Secrets Manager integration with `@ConfigurationProperties` is unclear

### DISCUSSION THREAD

**Discussion by ofir-popowski** [awspring/spring-cloud-aws#1198] [Q&A]
The documentation states:
> Fetched secrets can be referenced with `@Value`, bound to `@ConfigurationProperties` classes, or referenced in `application.properties` file.

but there's no example or explanation how this works. Say I have the following
```kotlin
@ConfigurationProperties("custom")
data class CustomProperties(
  val name: String,
  val key: String
)
```
my `application.properties` looked like this
```
custom.name=name
custom.key=key
```

Now I'm working on migrating those to secrets manager, and I have created a JSON secret under `test/custom`, and removed the previous properties and replaced them with `spring.config.import=aws-secretsmanager:test/custom`.

What else to I need to do to ensure that my configuration properties can be properly populated from this secret?

**Comment by maciejwalkowiak** [comment]
How does your JSON secret look like? There's nothing more you should do, this should work.

**Reply by ofir-popowski** [reply]
My JSON was:
```json
{
  "name": "name",
  "key": "key"
}
```

I forgot to update that after a couple of days I figured out that the prefix is mandatory, so `spring.config.import=aws-secretsmanager:test/custom?prefix=custom` worked
